mineinabyss-features/src/main/kotlin/com/mineinabyss/displayLocker/DisplayLockerFeature.kt

```diff
@@ -0,0 +1,84 @@
+import kotlinx.serialization.SerialName
+import net.minecraft.util.DirectoryLock.LockException
+import org.bukkit.Bukkit
+import org.bukkit.entity.Player
+
+
+@kotlinx.serialization.Serializable
+@SerialName("lock_display_item")
+class DisplayLockerFeature: AbyssFeature {
+    override fun MineInAbyssPlugin.enableFeature() {
+        registerEvents(DisplayLockerListener())
+
+        commands {
+            mineinabyss {
+                "lock"(desc="display locker commands") {
+                    val player = sender as Player
+                    val entity = player.playerData.recentRightclickedEntity
+                    val locked = entity?.toGeary()?.get<LockArmorStand>()
+
+                    "toggle"(desc="toggles locking on display item") playerAction@{
+                        locked?.lockState = !locked?.lockState!!
+                        entity.setGravity(!locked.lockState)
+                        entity.toGeary().encodeComponentsTo(entity)
+                        if (locked.lockState) player.success("this ${entity.name} is now protected!")
+                        else player.error("this ${entity.name} is no longer protected!")
+                    }
+                    "add"(desc="add a player to this display item") {
+                        val playerName by stringArg()
+                        val uuid = Bukkit.getOfflinePlayer(playerName).uniqueId
+
+                        if (!locked?.isAllowed(uuid)!!) {
+                            locked.allowedAccess.add(uuid)
+                            entity.toGeary().encodeComponentsTo(entity)
+                            player.success("$playerName is now able to interact with this ${entity.name}.")
+                        } else player.error("$playerName is already able to interact with this ${entity.name}")
+                    }
+                    "remove"(desc="remove a player from this display item") {
+                        val playerName by stringArg {parseErrorMessage = { "There is no player with the name: $passed." }}
+                        val uuid = Bukkit.getOfflinePlayer(playerName).uniqueId
+
+                        if (!locked?.isAllowed(uuid)!!) player.error("$playerName is unable to interact with this ${entity.name}.")
+                        else {
+                            locked.allowedAccess.remove(uuid)
+                            entity.toGeary().encodeComponentsTo(entity)
+                            player.success("$playerName was successfully removed from this ${entity.name}.")
+                        }
+                    }
+                    "clear"(desc="clear all other players from this display item") {
+                        locked?.allowedAccess?.clear()
+                        locked?.allowedAccess?.add(locked.owner)
+                        entity?.toGeary()?.encodeComponentsTo(entity)
+                        player.success("all other players were successfully removed from this ${entity?.name}.")
```

Comment:
yeah as above keep playerAction here and make sure that entity isnt null. cant encode components onto an entity that doesnt exist